### PR TITLE
Add Components General Guide and Factories

### PR DESCRIPTION
**Pull request by pjedrzejewski**
| Q             | A
| ------------- | ---
| Doc fix?      | yes
| New docs?     | yes (Sylius/Sylius#3613)
| Fixed tickets | -

I renamed 2 main chapters:

* Symfony2 Ecommerce Bundles -> Symfony Bundles
* PHP E-Commerce Components - PHP Components

Not everything is strictly e-commerce related.

Also added Components General Guide - which will cover common stuff for all components, so that we don't need to repeat it 324325 times. Same as for Bundles.

### CODE REVIEW

File: components/general/using_components.rst [hunk 1]
```diff
@@ -35,24 +35,33 @@ whatever component you want.
...
+immediately:
+
+.. code:: php
+
+    // Sample script.php file.
+
     <?php
 
-    // update this to the path to the "vendor/"
-    // directory, relative to this file
```
Comment (michalmarcinkowski) [inline components/general/using_components.rst]: I know it's obvious comment, but still it can be helpful, so why remove it?

**Comment by michalmarcinkowski** [comment]
:+1: for Components General Guide!

File: components/general/using_components.rst (hunk 1 above)
Comment (pjedrzejewski) [inline components/general/using_components.rst]: It just works if you follow docs and some may found it confusing as if they need to do something here, they don't. :) What do you think?

Comment (michalmarcinkowski) [inline components/general/using_components.rst]: Ok, let's remove it :)

**Comment by kzap** [comment]
the docs should contain why createNew() is better than just extending and using contructors

a concrete code example would help

also all docs should use createNew() and factories instead of new Class() as thats just confusing that the docs are not the way the code should be used.... if people will copy and paste to get started, they better copy and paste the right way, the factory is an opinion and its best to show them that way

**Pull request merged**